ghost/core/core/server/services/themes/validate.js

```diff
@@ -5,20 +5,39 @@ const config = require('../../../shared/config');
 const labs = require('../../../shared/labs');
 const tpl = require('@tryghost/tpl');
 const errors = require('@tryghost/errors');
+const adapterManager = require('../adapter-manager');
+const logging = require('@tryghost/logging');
+const list = require('./list');
 
 const messages = {
     themeHasErrors: 'Theme "{theme}" is not compatible or contains errors.',
     activeThemeHasFatalErrors: 'The currently active theme "{theme}" has fatal errors.',
-    activeThemeHasErrors: 'The currently active theme "{theme}" has errors, but will still work.'
+    activeThemeHasErrors: 'The currently active theme "{theme}" has errors, but will still work.',
+    themeNotLoaded: 'Theme "{themeName}" is not loaded and cannot be checked.'
+};
+
+/**
+ * @typedef {Object} CacheStore
+ * @property {(key: string) => Promise<any>} get - get a value from the cache. Returns undefined if not found
+ * @property {(key: string, value: any) => Promise<void>} set - set a value in the cache
+ * @property {() => Promise<void>} reset - reset the cache
+ */
+
+/**
+ * The cache store for storing the result of the last theme validation
+ * @type {CacheStore}
+ */
+let gscanCacheStore;
+
+module.exports.init = () => {
+    gscanCacheStore = adapterManager.getAdapter('cache:gscan');
```

Comment:
I think this way of initializing the adapter is fine for now, given the time constraint. But we should leave a comment and open an issue to redo this initialization into calss with "cacheStore" passed in as a parameter. Validator should now know anything about adapter manager ideally.

I see we only call "init" once anyway, maybe worth passing `gscanCacheStore` as a parameter straight away.

Reply:
Oh good to know! I wasn't sure about this as I only found usages of `adapterManager.getAdapter('cache:*')`